File: Editorial/0136_Single_Number/CSharp/2_Hash_Table.cs
public class Solution {
    public int SingleNumber(int[] nums) {
        var hashTable = new Dictionary<int, int>();
        foreach (int num in nums) {
            if (hashTable.ContainsKey(num)) {
                hashTable[num]++;
            } else {
                hashTable[num] = 1;
            }
        }

        foreach (int num in nums) {
            if (hashTable[num] == 1) {
                return num;
            }
        }

        return 0;
    }
}

File: Editorial/0136_Single_Number/CPP/2_Hash_Table.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int> hash_table;
        for (int num : nums) {
            hash_table[num]++;
        }
        for (int num : nums) {
            if (hash_table[num] == 1) {
                return num;
            }
        }
        return 0;
    }
};
